framework/src/mesh/SpiralAnnularMesh.C

```diff
@@ -0,0 +1,306 @@
+
+  if (_second_order)
+  // if (getParam<bool>("second_order"))
+  {
+    mesh.all_second_order(/*full_ordered=*/true);
+    std::vector<unsigned int> nos;
+
+    // Loop over the elements, moving mid-edge nodes onto the
+    // nearest radius as applicable. For each element, exactly one
+    // edge should lie on the same radius, so we move only that
+    // mid-edge node.
+    for (const auto & elem : mesh.element_ptr_range())
+    {
+      // Make sure we are dealing only with triangles
+      libmesh_assert(elem->n_vertices() == 3);
+
+      // Compute vertex radii
+      Real radii[3] = {elem->point(0).norm(), elem->point(1).norm(), elem->point(2).norm()};
+
+      // Compute absolute differences between radii so we can determine which two are on the same
+      // circular arc.
+      Real dr[3] = {std::abs(radii[0] - radii[1]),
+                    std::abs(radii[1] - radii[2]),
+                    std::abs(radii[2] - radii[0])};
+
+      // Compute index of minimum dr.
+      auto index = std::distance(std::begin(dr), std::min_element(std::begin(dr), std::end(dr)));
+
+      // Make sure that the minimum found is also (almost) zero.
+      if (dr[index] > TOLERANCE)
+        mooseError("Error: element had no sides with nodes on same radius.");
+
+      // Get list of all local node ids on this side. The first
+      // two entries in nos correspond to the vertices, the last
+      // entry corresponds to the mid-edge node.
+      nos = elem->nodes_on_side(index);
+
+      // Compute the angles associated with nodes nos[0] and nos[1].
+      Real theta0 = std::atan2(elem->point(nos[0])(1), elem->point(nos[0])(0)),
+           theta1 = std::atan2(elem->point(nos[1])(1), elem->point(nos[1])(0));
+
+      // atan2 returns values in the range (-pi, pi).  If theta0
+      // and theta1 have the same sign, we can simply average them
+      // to get half of the acute angle between them. On the other
+      // hand, if theta0 and theta1 are of opposite sign _and_ both
+      // are larger than pi/2, we need to add 2*pi when averaging,
+      // otherwise we will get half of the _obtuse_ angle between
+      // them, and the point will flip to the other side of the
+      // circle (see below).
+      Real new_theta = 0.5 * (theta0 + theta1);
+
+      // It should not be possible for both:
+      // 1.) |theta0| > pi/2, and
+      // 2.) |theta1| < pi/2
+      // as this would not be a well-formed element.
+      if ((theta0 * theta1 < 0) && (std::abs(theta0) > 0.5 * libMesh::pi) &&
+          (std::abs(theta1) > 0.5 * libMesh::pi))
+        new_theta = 0.5 * (theta0 + theta1 + 2 * libMesh::pi);
+
+      // The new radius will be the radius of point nos[0] or nos[1] (they are the same!).
+      Real new_r = elem->point(nos[0]).norm();
+
+      // Finally, move the point to its new location.
+      elem->point(nos[2]) = Point(new_r * std::cos(new_theta), new_r * std::sin(new_theta), 0.);
+    }
+  }
+
+  mesh.write("bassi_rebay_mesh.e");
```

Comment:
This line needs to be removed! We wrote the mesh in the original test code but don't want to do that here.